_includes/_code/clickbox-c/main.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+void main() {
+  runApp(new MyApp());
+}
+
+//---------------------------- ParentWidget ----------------------------
+
+class ParentWidget extends StatefulWidget {
+  @override
+  ParentWidgetState createState() => new ParentWidgetState();
+}
+
+class ParentWidgetState extends State<ParentWidget> {
+  bool active = false;
+
+  void handleClickboxChanged(bool newValue) {
+    setState(() {
+      active = newValue;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Container(
+      child: new ClickboxC(
+        active: active,
+        onChanged: handleClickboxChanged,
+      ),
+    );
+  }
+}
+
+//----------------------------- ClickboxC ------------------------------
+
+class ClickboxC extends StatefulWidget {
+  ClickboxC({Key key, this.active: false, @required this.onChanged})
+      : super(key: key);
+
+  final bool active;
+  final ValueChanged<bool> onChanged;
+
+  ClickboxCState createState() => new ClickboxCState();
+}
+
+class ClickboxCState extends State<ClickboxC> {
+  bool highlight = false;
+
+  void handleTapDown(TapDownDetails details) {
+    setState(() {
+      highlight = true;
+    });
+  }
+
+  void handleTapUp(TapUpDetails details) {
+    setState(() {
+      highlight = false;
+    });
+  }
+
+  void handleTapCancel() {
+    setState(() {
+      highlight = false;
+    });
+  }
+
+  void handleTap() {
+    config.onChanged(!config.active);
+  }
+
+  Widget build(BuildContext context) {
+    // This example adds a green border on tap down.
+    // On tap up, the square changes to the opposite state.
+    return new GestureDetector(
+      onTapDown: handleTapDown, // Handle the tap events in the order that
```

Comment:
this comment "in the order that they occur" didn't make much sense to me. The order of the named parameters doesn't affect the order in which they are handled, AFAIK